Sources/TuistKit/Services/RunService.swift

```diff
@@ -0,0 +1,142 @@
+import Foundation
+import RxSwift
+import TSCBasic
+import TuistAutomation
+import TuistCore
+import TuistSupport
+
+enum RunServiceError: FatalError {
+    case appNotFound(targetName: String)
+    case simulatorNotFound
+    case targetNotFound(forSchemeName: String)
+
+    var description: String {
+        switch self {
+        case let .appNotFound(targetName):
+            return "Couldn't find any apps built for \(targetName)"
+        case .simulatorNotFound:
+            return "Could't find simulator"
+        case let .targetNotFound(forSchemeName):
+            return "Couldn't find \(forSchemeName)"
+        }
+    }
+
+    var type: ErrorType {
+        switch self {
+        case .appNotFound, .simulatorNotFound, .targetNotFound:
+            return .abort
+        }
+    }
+}
+
+final class RunService {
+    let simulatorController: SimulatorControlling
+
+    let buildGraphInspector: BuildGraphInspecting
+
+    let projectGenerator: ProjectGenerating
+
+    let xcodebuildController: XcodeBuildControlling
+
+    var appPath: AbsolutePath?
+
+    var buildableTarget: Target?
+
+    init(simulatorController: SimulatorControlling = SimulatorController(),
+         buildGraphInspector: BuildGraphInspecting = BuildGraphInspector(),
+         projectGenerator: ProjectGenerating = ProjectGenerator(),
+         xcodebuildController: XcodeBuildControlling = XcodeBuildController())
+    {
+        self.simulatorController = simulatorController
+        self.buildGraphInspector = buildGraphInspector
+        self.projectGenerator = projectGenerator
+        self.xcodebuildController = xcodebuildController
+    }
+
+    func run(schemeName: String) throws {
+        try buildScheme(schemeName)
+
+        guard let buildableTarget = buildableTarget else {
+            throw RunServiceError.targetNotFound(forSchemeName: schemeName)
+        }
+
+        appPath = try findApp(for: buildableTarget)
+        try launchApp(for: buildableTarget)
+    }
+
+    private func buildScheme(_ schemeName: String) throws {
+        let graph = try projectGenerator.load(path: FileHandler.shared.currentPath)
+        guard let scheme = buildGraphInspector.buildableSchemes(graph: graph).first(where: { $0.name == schemeName }) else {
+            throw RunServiceError.appNotFound(targetName: "")
```

Comment:
Should the error here be `RunServiceError.schemeNotFound(scheme: schemeName, availableSchemes: schemes)`?